### PR TITLE
A keyword should be followed by a non-alphanumberic character

### LINKED ISSUE #203: Scanner failed
```
if(true)
  print(1);
else
  print(2);
```

### PR DESCRIPTION
**Pull request by lenguyenthanh**
close #203

### CODE REVIEW

**Comment by lenguyenthanh** [comment]
cc @grokking-vietnam/lab-fundamentals-programing-principle

File: compiler/shared/src/main/scala/grox/Scanner.scala [hunk 1]
```diff
@@ -55,13 +57,9 @@ object Scanner:
 
   // An identifier can only start with an undercore or a letter
```
Comment (hungcaox) [inline compiler/shared/src/main/scala/grox/Scanner.scala:58]: typo: underscore

File: compiler/shared/src/main/scala/grox/Scanner.scala [hunk 2]
```diff
@@ -55,13 +57,9 @@ object Scanner:
...
   // and can contain underscore or letter or numeric character
-  val identifier: P[Token[Unit]] =
-    val alphaOrUnderscore = R.alpha | P.char('_')
-    val alphaNumeric = alphaOrUnderscore | N.digit
-
-    (alphaOrUnderscore ~ alphaNumeric.rep0)
-      .string
-      .map(Identifier(_, ()))
+  val identifier: P[Token[Unit]] = (alphaOrUnderscore ~ alphaNumeric.rep0)
```
Comment (hungcaox) [inline compiler/shared/src/main/scala/grox/Scanner.scala:60]: 💯

File: compiler/shared/src/main/scala/grox/Scanner.scala [hunk 3]
```diff
@@ -126,7 +124,11 @@ object Scanner:
 
   extension (t: Token[Unit])
     def operator = P.string(t.lexeme).as(t)
-    def keyword = (P.string(t.lexeme).as(t) <* (whitespace | P.end).peek).backtrack
+
+    // The character that immediately follows a keyword must not be a alphanumberic character.
```
Comment (hungcaox) [inline compiler/shared/src/main/scala/grox/Scanner.scala:128]: Is it comment necessary given that the sentence at line 130 describes the same thing?

Comment (hungcaox) [reply compiler/shared/src/main/scala/grox/Scanner.scala:128]: nit: extra line 129.

**Review by hungcaox — APPROVED** [review]

**Comment by lenguyenthanh** [comment]
thanks @hungcaox !

**Pull request merged**